app/apollo/resolvers/common.js

```diff
@@ -55,53 +55,76 @@ const validClusterAuth = async (me, queryName, context) => {
   }
 };
 
-var getAllowedChannels = async(me, org_id, action, field, context)=>{
+// Polymorphically check for cached IAM decision, Get resources authorized by Access Policy, Update cache for individual resource authentication
+var getAllowedResources = async(me, org_id, action, type, queryName, context, searchByTags=null, searchByUuid=null, searchQuery=null)=>{
   const { models } = context;
-  var channels = await models.Channel.find({ org_id });
-  return await filterChannelsToAllowed(me, org_id, action, field, channels, context);
-};
 
-var filterChannelsToAllowed = async(me, org_id, action, field, channels, context)=>{
-  const { models } = context;
-  var decisionInputs = _.map(channels, (channel)=>{
-    return {
-      type: field,
-      action,
-      uuid: channel.uuid,
-      name: channel.name,
-    };
-  });
-  var decisions = await models.User.isAuthorizedBatch(me, org_id, decisionInputs, context);
-  channels = _.filter(channels, (val, idx)=>{
-    return decisions[idx];
-  });
-  return channels;
-};
+  var resources;
 
-var getAllowedSubscriptions = async(me, org_id, action, field, context)=>{
-  const { models } = context;
-  var subscriptions = await models.Subscription.find({ org_id });
-  return await filterSubscriptionsToAllowed(me, org_id, action, field, subscriptions, context);
+  // Check for cached IAM decision, returns true if cache is empty or all is authorized
+  const allAllowed = await cacheAllAllowed(me, org_id, action, type, queryName, context);
```

Comment:
If the cache **doesn't** say "user is authorized to ALL of type XYZ", then the code would continue to query the database, do a fine-grained authorization check against each resource found from the DB query, cache the fine grained auth decision, and return the filtered list of resources.

Doesn't this mean that the cache will **never** have a 'hit' for "user is authorized to ALL of type XYZ"?  There's nothing doing a query for "auth to ALL" and caching the result.

Which would make this cacheAllAllowed function.... useless?

Reply:
The issue here was like the other review comment on this function. A lack of an update to my comments. This is resolved

Reply:
The updates to the comments make it more clear what is happening, but doesn't address this part of my comment:

> Doesn't this mean that the cache will never have a 'hit' for "user is authorized to ALL of type XYZ"? There's nothing doing a query for "auth to ALL" and caching the result.
> 
> Which would make this cacheAllAllowed function.... useless?

Reply:
Changes just pushed up in [convert cacheAllAllowed to validAuth] commit.